destination/src/main/scala/quasar/plugin/sqlserver/destination/SQLServerDestination.scala

```diff
@@ -64,6 +73,17 @@ private[destination] final class SQLServerDestination[F[_]: ConcurrentEffect: Mo
 }
 
 object SQLServerDestination {
+  private def retryN[F[_]: Effect: Timer](maxN: Int, timeout: FiniteDuration, n: Int): ConnectionIO ~> ConnectionIO =
+    λ[ConnectionIO ~> ConnectionIO] { action =>
+      action.attempt flatMap {
+        case Right(a) => a.pure[ConnectionIO]
+        case Left(e) if n < maxN =>
+          toConnectionIO[F].apply(Timer[F].sleep(timeout)) >>
+          retryN[F](maxN, timeout, n + 1).apply(action)
```

Comment:
Maybe we retry `rollback >> action` as we aren't restricting to loss-of-connection failures here?

Reply:
:+1: